### DISCUSSION TITLE
How do you retrieve mos parameters?

### DISCUSSION THREAD

**Discussion by medwatt** [ra3xdh/qucs_s#357] [Q&A] [answered]
In ngpsice, let's say you want to know the `gm` of the mosfet called `m1`,  you can do so by writing `save @m1[gm]`. How do you do that in qucs?

**Comment by ra3xdh** [comment]
You have to use script simulation type if you need to extract variables that are nor voltage nor current or its dependencies. This simulation type doesn't support the `save` command. Use `write` instead. You can find some hints how to use this simulation type here: https://mos-ak.org/lausanne_2017/presentations/T03_Kuznetsov_MOS-AK_DATE_2017.pdf Sorry, there is no better documentation.

**Reply by medwatt** [reply]
Thanks for your reply. I did the simulation using `Nutmeg script`. This netlist that was generated is shown below. It simulated fine. I could see all the parameters in the tabular diagram and I could get the values of the node voltages and currents. However, while `@m1[gm]` is available, it shows `no data` in the diagram.

```spice
* Qucs 2.0.0 /home/medwatt/.qucs/test_prj/common_source_testing.sch
.INCLUDE "/home/medwatt/qucs-s/share/qucs-s/xspice_cmlib/include/ngspice_mathfunc.inc"
* Qucs 2.0.0  /home/medwatt/.qucs/test_prj/common_source_testing.sch
.INCLUDE "/home/medwatt/models/NMOS_VTL.inc"
V2 _net0 0 DC 1
M1 out  in  0  0 NMOS_VTL L=75n W=3.25u
V3 in _net1 DC 0 SIN(0 0.1M 1K 0 0 0) AC 0.1M ACPHASE 0
C1 0 out  5P 
R1 out _net0  5K

V1 _net1 0 DC 0.357
.control
echo "" > spice4qucs.cir.noise
echo "" > spice4qucs.cir.pz
set filetype=ascii
save all @m1[id] @m1[vth] @m1[gm]
op
write common_source_testing_custom.txt all
destroy all
reset
exit
.endc
.END
```

**Comment by ra3xdh** [comment]
You can also try to specify the equation dependent on device parameter as the Nutmeg equation. Keep in mind that Qucs-S may add letter (i.e M,Q,D) to the device name to ensure the SPICE compatibility. You may require to look into the netlist to find exact device reference.

**Comment by ra3xdh** [comment]
Everything is correct. You have got a set of 1x1 point plots. You may look at it only in tabular form.

**Reply by medwatt** [reply]
Maybe, my post was not clear. I am using tabular form. It shows `no data`.

![image](https://github.com/ra3xdh/qucs_s/assets/17733465/3382f8bb-de92-4e76-837b-39ff8e4371dd)

**Comment by ra3xdh** [comment]
You may look at #277 how to assemble a set of 1x1 points into one curve.

**Comment by ra3xdh** [comment]
Yes, I understood now. The `@` symbol is reserved for `PlotVs()` emulation. Try to assign a variable `let m1_gm=@m1[gm]` and plot it instead of plotting `@m1[gm]` directly.

**Reply by medwatt** [reply]
This doesn't work in dc sweeps. You can see in the table below that it only returns the value in the first step value. Using `save @m1[vth]` doesn't have this problem.

![image](https://github.com/ra3xdh/qucs_s/assets/17733465/bccbec8b-0670-44fc-b180-b13d959ecb62)

**Comment by ra3xdh** [comment, marked as answer]
After some experimenting I have found the following approach how to use `save` with the Nutmeg script and bypass the reserved `@` symbol. Look at the example schematic. The result is identical to CLI simulation.
![image](https://github.com/ra3xdh/qucs_s/assets/4920080/695ddfe3-cece-4ba5-8df5-d0d9bbc8bd62)
![image](https://github.com/ra3xdh/qucs_s/assets/4920080/329b70a4-b8d4-4034-9e2f-fd1529f8c7c1)
